fix checkbox width -  if autosize True;

blackholeearth opened the pull request.

jpobst commented:
Unfortunately this is not the way to implement AutoSize correctly.  You will need for `CheckBox` to override `Control.GetPreferredSizeCore` and calculate a new size that takes every possible size modifier into account, eg:

- Font
- Text
- Check glyph
- Image
- ImageTextRelation
- Border width
- Padding

blackholeearth commented:
current state

https://github.com/user-attachments/assets/69d1b1bd-8e75-465f-826c-48fed0cee802

blackholeearth commented:
also textalign , imagealign , glypalign ,, after these changed..

need to call trigger autosize . othwerwise . the control are may get bigger on each change in these enums.

------
i realized radiobutton also has the same problem

jpobst commented:
> i realized you have written GetPreferredSizeCore . if its wrong, i can move the code there.
but it work inside ""GetPreferredSize"" ??

Overriding `Control.GetPreferredSize` breaks all the AutoSize logic in there.  If you override `Control.GetPreferredSizeCore` then you shouldn't need to call anything in the property setters like `TextAlign` (as long as they already have the `LayoutTransaction.DoLayoutIf (AutoSize, Parent, this, PropertyNames.TextAlign)` call).

> i realized radiobutton also has the same problem

Yes, the `Size calculate_AutosizeArea ()` code probably needs to go into `TextImageLayoutEngine.GetPreferredSize (Control control, Size proposedSize)`.  Then it can be called by all the Controls that use `TextImageLayoutEngine` like `Button`, `Label`, `RadioButton`, and `CheckBox`.

blackholeearth commented:
i moved calculate_AutosizeArea to GetPrefferredSizeCore

it Works as same..

2) 

i removed trigger resizing code..

but if i do that,
on every Enum change  (TextRealation etc)

checkbox controls gets bigger…

see here-> video

https://github.com/user-attachments/assets/77d48c31-cd8c-469e-903d-19f906856b8f



---
* i have to keep triggerResizing() as temporaryHack..
 
 otherwise chekbox size gets bigger bigger on every enum change.



---
edit:


currently in ``calculate_AutosizeArea``   
i am using ``TextImageLayoutData.layout()``   //just like you do in the  ChekboxRenderer..

but the TextImageLayoutData.layout  


i do calculation believing that 
info result of  ``TextImageLayoutData.layout()``  is corrent ??

jpobst commented:
I don't think `TextImageLayoutData.Layout ()` is going to work for this purpose.  It is designed to fit the content the best it can within a predetermined size.

You need the opposite: what is the ideal size to fit the content?

blackholeearth commented:
instead of x.y posititons  ..

latest code  uses engine.layout() . widths  Heights.

now enums dont need trigger resizing.
